compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -477,7 +477,7 @@ object Erasure {
 	 */
     override def typedApply(tree: untpd.Apply, pt: Type)(implicit ctx: Context): Tree = {
       val Apply(fun, args) = tree
-      if (fun.symbol == defn.cbnArg)
+      if (fun.symbol == defn.cbnArg || (tree.symbol.name == nme.box || tree.symbol.name == nme.unbox) && tree.symbol.owner.asClass.companionClass == defn.UnitClass)
```

Comment:
I would factor out the newly added tests into an inner function, to make it clear what is tested: `def isUnitBoxOrUnbox(fun: Symbol)`